### PR TITLE
CDAP-3074 java versions to support Java 8

### PR DESCRIPTION
**Pull request by rohanahata**
Updates shell scripts to use java 8.

### CODE REVIEW

**Comment by wolf31o2** [comment]
:+1: LGTM

**Comment by awholegunch** [comment]
-1 There are typos: please don't merge until I've finished looking

File: cdap-cli/bin/cdap-cli.bat [hunk 1]
```diff
@@ -48,8 +48,10 @@ set line=%line:java version "1.=!!%
 set line=%line:~0,1%
 if NOT "%line%" == "6" (
   if NOT "%line%" == "7" (
-    echo ERROR: Java version not supported. Please install Java 6 or 7 - other versions of Java are not yet supported.
-    GOTO :FINALLY
+    if NOT "%line%" == "8" (
+      echo ERROR: Java version not supported. Please install Java 6, 7 or 8 - other versions of Java are not yet supported.
```
Comment (awholegunch) [inline cdap-cli/bin/cdap-cli.bat]: Java 6, 7, or 8: other versions of Java are not yet supported.

File: cdap-data-fabric/bin/tx-debugger.bat [hunk 2]
```diff
@@ -49,8 +49,10 @@ set line=%line:java version "1.=!!%"
 set line=%line:~0,1%
 if NOT "%line%" == "6" (
   if NOT "%line%" == "7" (
-    echo ERROR: Java version not supported. Please install Java 6 or 7 - other versions of Java are not yet supported.
-    GOTO :FINALLY
+    if NOT "%line%" == "8" (
+      echo ERROR: Java version not supported. Please install Java 6, 7 or 8 - other versions of Java are not yet supported.
```
Comment (awholegunch) [inline cdap-data-fabric/bin/tx-debugger.bat]: Java 6, 7, or 8: other versions of Java are not yet supported.

File: cdap-standalone/bin/cdap.bat [hunk 3]
```diff
@@ -100,8 +100,10 @@ set line=%line:java version "1.=!!%
 set line=%line:~0,1%
 if NOT "%line%" == "6" (
   if NOT "%line%" == "7" (
-    echo ERROR: Java version not supported. Please install Java 6 or 7 - other versions of Java are not yet supported.
-    GOTO :FINALLY
+    if NOT "%line%" == "8" (
+      echo ERROR: Java version not supported. Please install Java 6, 7 or 8 - other versions of Java are not yet supported.
```
Comment (awholegunch) [inline cdap-standalone/bin/cdap.bat]: Java 6, 7, or 8: other versions of Java are not yet supported.

File: cdap-standalone/bin/cdap.sh [hunk 4]
```diff
@@ -98,9 +98,9 @@ fi
 
 # java version check
 JAVA_VERSION=`$JAVACMD -version 2>&1 | grep "java version" | awk '{print $3}' | awk -F '.' '{print $2}'`
-if [ $JAVA_VERSION -ne 6 ] && [ $JAVA_VERSION -ne 7 ]; then
+if [ $JAVA_VERSION -ne 6 ] && [ $JAVA_VERSION -ne 7 ] && [ $JAVA_VERSION -ne 8 ]; then
   die "ERROR: Java version not supported
-Please install Java 6 or 7 - other versions of Java are not yet supported."
+Please install Java 6, 7 or 8 - other versions of Java are not yet supported."
```
Comment (awholegunch) [inline cdap-standalone/bin/cdap.sh]: Java 6, 7, or 8: other versions of Java are not supported.

**Comment by awholegunch** [comment]
I'm thinking we should say "not supported" instead of "not yet", as we don't want to imply we are going to support any other versions.

**Comment by awholegunch** [comment]
And shouldn't all these say "Java 7 or 8"? The docs no longer mention Java 6...

File: cdap-cli/bin/cdap-cli.bat [hunk 5]
```diff
@@ -48,8 +48,10 @@ set line=%line:java version "1.=!!%
 set line=%line:~0,1%
 if NOT "%line%" == "6" (
```
Comment (awholegunch) [inline cdap-cli/bin/cdap-cli.bat]: If we are no longer supporting 6, this needs revising...

File: cdap-standalone/bin/cdap.sh [hunk 6]
```diff
@@ -98,9 +98,9 @@ fi
 
 # java version check
 JAVA_VERSION=`$JAVACMD -version 2>&1 | grep "java version" | awk '{print $3}' | awk -F '.' '{print $2}'`
-if [ $JAVA_VERSION -ne 6 ] && [ $JAVA_VERSION -ne 7 ]; then
+if [ $JAVA_VERSION -ne 6 ] && [ $JAVA_VERSION -ne 7 ] && [ $JAVA_VERSION -ne 8 ]; then
```
Comment (awholegunch) [inline cdap-standalone/bin/cdap.sh]: If we don't support 6, this needs to be revised.

**Comment by awholegunch** [comment]
Since we don't support Java 6 anymore, this PR needs to be revised...might be easier to start over, but with the changes mentioned above in mind.

**Comment by wolf31o2** [comment]
:+1: LGTM... I don't know what typos there were. I see none. Anyway, this looks good, assuming @awholegunch approves. :smile_cat:

**Comment by awholegunch** [comment]
@wolf31o2 Typo was "6, 7 or 8" should have been "6, 7, or 8"...but now LGTM...

**Comment by wolf31o2** [comment]
Meh... Oxford comma

**Pull request merged**